elastic/apm-server
# Reintroduce integration package generator

## Motivation/summary

Introduce a new integration package generator/transformer, which takes a partial integration package in apmpackage/apm and performs APM-specific transformations and validations:

 - Inject package version and Kibana version constraint
 - Inject common ingest pipelines
 - Check ilm_policy is present and correct (and fix the
   value in the sampled_traces)
 - Copy fields from traces to rum_traces

We no longer explicitly set `event.ingested` in our ingest pipelines; this is done by the common Fleet-installed pipeline, which is set as the final pipeline in all Fleet-install templates.

## Checklist

~- [ ] Update [CHANGELOG.asciidoc](https://github.com/elastic/apm-server/blob/master/CHANGELOG.asciidoc)~
~- [ ] Documentation has been updated~

## How to test these changes

1. Install the package
2. Check all index templates' ILM policy references are valid

## Related issues

Prereq for https://github.com/elastic/apm-server/issues/6425
